Add a guide to Proxy Service traffic separation

ptgott opened the pull request:
Closes #27885
Closes #19276

Add a guide to the Self-Hosting Teleport section laying out the options described in #27885 for separating internal and external Proxy Service traffic.

ptgott committed:
Add a guide to Proxy Service traffic separation

Closes #27885

Add a guide to the Self-Hosting Teleport section laying out the options
described in #27885 for separating internal and external Proxy Service
traffic.

pschisa commented on docs/pages/admin-guides/deploy-a-cluster/separate-proxy-service-endpoints.mdx:
```diff
@@ -0,0 +1,184 @@
+   address listed, to use the value of the environment variable instead.
+
+   The step differs according to whether you installed Teleport on Linux servers
+   using a package manager or using the `teleport-kube-agent` Helm chart:
```
Would it be worth mentioning managed upgrades v2 since we moved away from package managers?

pschisa commented on docs/pages/admin-guides/deploy-a-cluster/separate-proxy-service-endpoints.mdx:
```diff
@@ -0,0 +1,184 @@
+     enabled: true
+     public_addr: 
+       - external_proxy.example.com:3080
+       - internal_proxy.example.com:3024
```
The tunnel endpoint generally still uses port :3080 (or more commonly 443) because in this example multiplexing is still enabled and there is no separate tunnel listener.

ptgott committed:
Respond to pschisa feedback

- Make it more explicit that the guide applies to self-hosted clusters.
- Mention Managed Updates v2 instead of package managers.
- Fix example ports in the TELEPORT_TUNNEL_PUBLIC_ADDR section.
- Mention DNS in the "Multiple public addresses" section

hugoShaka reviewed (commented):
I wrote a comment last week and forgot to send it 🫠

hugoShaka commented on docs/pages/admin-guides/deploy-a-cluster/separate-proxy-service-endpoints.mdx:
````diff
@@ -0,0 +1,196 @@
+       - internal_proxy.example.com:443
+   ```
+
+1. On Teleport Agent hosts, start the `teleport` process with the
+   `TELEPORT_TUNNEL_PUBLIC_ADDR` environment variable assigned to the internal
+   Proxy Service load balancer address. 
+
+   Setting `TELEPORT_TUNNEL_PUBLIC_ADDR` overrides the default behavior, in
+   which an agent verifies the Proxy Service certificate using the first public
+   address listed, to use the value of the environment variable instead.
````
Yeah, just forget my comment. It's better bto document this than to use an undocumented environment variable in real deployments.

hugoShaka commented on docs/pages/admin-guides/deploy-a-cluster/separate-proxy-service-endpoints.mdx:
```diff
@@ -0,0 +1,196 @@
+- [Multiple public addresses](#multiple-public-addresses)
+- [Tunnel public address](#tunnel-public-address)
+
+### Split DNS
+
+You can set up your infrastructure to use split DNS so that a single public
+address for your Teleport cluster resolves to the external endpoint for end
+users and the internal endpoint for infrastructure resources in your internal
+network.
```
Can we say that this is the _strongly recommended_ option. Option #2 requires setting an env var on every agent which is cumbersome, and error-prone. Option #3 has the users disable TLS multiplexing which is the opposite of what we want, we are currently strongly pushing for every deployment to use TLS multiplexing as this heavily reduces the risks of something going wrong in the network setup, is easier to troubleshoot, and offers better/faster/more reliable connectivity.

pschisa replied:
Agree with @hugoShaka here.  Split DNS to me is the easiest and cleanest approach and should always be the first option presented.  The others are only there because the concept of split DNS is polarizing with customers at times.

ptgott committed:
Respond to feedback

Edit the Proxy Service traffic separation guide to include a strong
recommendation to use the split DNS approach. Edit explanations of other
approaches to frame them as contingencies for if a cluster doesn't meet
the requirements of the split DNS approach.
